### ISSUE TITLE
Exception: Cannot read all data. Bytes read: 0. Bytes expected: 1

### ISSUE THREAD

**Issue by DBalashov** [DarkWanderer/ClickHouse.Client#21]
Test project (net fw 4.8) & data: https://i.tk-chel.ru/denisio/TestCH.zip

1. Unpack, create new database and table from table.sql file.
2. Edit Program.cs - connection string & path to 1.bin (msgpack serialized test data, 10K rows)
3. Run. Exception will occured. Clickhouse version 20.1.2.4 (official build)

```
            using (var bc = new ClickHouseBulkCopy(c) {DestinationTableName = "Tabular"})
                try
                {
                    var values = data.data.Select(p => p.items).ToArray();

                    await bc.WriteToServerAsync(values, data.colNames);
                }
                catch (Exception e)
                {
                    Console.WriteLine((e.InnerException ?? e).Message);
                }

```
clickhouse-server.err.log :

> 2020.03.28 11:26:47.802202 [ 46 ] {be18b95d-8cf2-44f2-af75-63cb9007a466} <Error> executeQuery: Code: 33, e.displayText() = DB::Exception: Cannot read all data. Bytes read: 0. Bytes expect
> ed: 1. (version 20.1.2.4 (official build)) (from [::ffff:192.168.0.10]:12205) (in query: INSERT INTO Tabular FORMAT RowBinary ), Stack trace (when copying this message, always include the
>  lines below):
> 
> 0. 0xbc31d9c Poco::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)  in /usr/bin/clickhouse
> 1. 0x4f6ccd9 DB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)  in /usr/bin/clickhouse
> 2. 0x4b6efee ?  in /usr/bin/clickhouse
> 3. 0x91ebf5f DB::BinaryRowInputFormat::readRow(std::__1::vector<COW<DB::IColumn>::mutable_ptr<DB::IColumn>, std::__1::allocator<COW<DB::IColumn>::mutable_ptr<DB::IColumn> > >&, DB::RowRea
> dExtension&)  in /usr/bin/clickhouse
> 4. 0x97e5f69 DB::IRowInputFormat::generate()  in /usr/bin/clickhouse
> 5. 0x91a4c27 DB::ISource::work()  in /usr/bin/clickhouse
> 6. 0x9169435 DB::InputStreamFromInputFormat::readImpl()  in /usr/bin/clickhouse
> 7. 0x8a6d32f DB::IBlockInputStream::read()  in /usr/bin/clickhouse
> 8. 0x8bec745 DB::InputStreamFromASTInsertQuery::readImpl()  in /usr/bin/clickhouse
> 9. 0x8a6d32f DB::IBlockInputStream::read()  in /usr/bin/clickhouse
> 10. 0x8a9e66b DB::copyData(DB::IBlockInputStream&, DB::IBlockOutputStream&, std::__1::atomic<bool>*)  in /usr/bin/clickhouse
> 11. 0x8dc3cc4 DB::executeQuery(DB::ReadBuffer&, DB::WriteBuffer&, bool, DB::Context&, std::__1::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocato
> r<char> > const&)>, std::__1::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)>)  in /usr/bin/clickhouse
> 12. 0x4ffdb8f DB::HTTPHandler::processQuery(Poco::Net::HTTPServerRequest&, HTMLForm&, Poco::Net::HTTPServerResponse&, DB::HTTPHandler::Output&)  in /usr/bin/clickhouse
> 13. 0x5001a73 DB::HTTPHandler::handleRequest(Poco::Net::HTTPServerRequest&, Poco::Net::HTTPServerResponse&)  in /usr/bin/clickhouse
> 14. 0x9bb8bec Poco::Net::HTTPServerConnection::run()  in /usr/bin/clickhouse
> 15. 0x9bb5677 Poco::Net::TCPServerConnection::start()  in /usr/bin/clickhouse
> 16. 0x9bb5a6d Poco::Net::TCPServerDispatcher::run()  in /usr/bin/clickhouse
> 17. 0xbca44ff Poco::PooledThread::run()  in /usr/bin/clickhouse
> 18. 0xbca15c8 Poco::ThreadImpl::runnableEntry(void*)  in /usr/bin/clickhouse
> 19. 0xbca2e69 ?  in /usr/bin/clickhouse
> 20. 0x76db start_thread  in /lib/x86_64-linux-gnu/libpthread-2.27.so
> 21. 0x12188f clone  in /lib/x86_64-linux-gnu/libc-2.27.so
>

**Comment by DarkWanderer** [comment]
Thanks for good repro. Seems to be a problem with ClickHouseBulkCopy - I forgot to add column names to INSERT statement, so ClickHouse expects to have full row. Committed a provisional fix, will add more tests around it & release

**Comment by DarkWanderer** [comment]
Fixed in [1.1.0.197](https://github.com/DarkWanderer/ClickHouse.Client/releases/tag/1.1.0.197)
